lib/chef/knife/windows_cert_generate.rb

```diff
@@ -0,0 +1,158 @@
+# Author:: Mukta Aphale (<mukta.aphale@clogeny.com>)
+# Copyright:: Copyright (c) 2014-2016 Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require_relative "../knife"
+require_relative "winrm_base"
+
+class Chef
+  class Knife
+    class WindowsCertGenerate < Knife
+
+      attr_accessor :thumbprint, :hostname
+
+      banner "knife windows cert generate FILE_PATH (options)"
+
+      deps do
+        require "openssl"
+        require "socket"
+      end
+
+      option :hostname,
+        short: "-H HOSTNAME",
+        long: "--hostname HOSTNAME",
+        description: "Use to specify the hostname for the listener.
+        For example, --hostname something.mydomain.com or *.mydomain.com.",
+        required: true
+
+      option :output_file,
+        short: "-o PATH",
+        long: "--output-file PATH",
+        description: "Specifies the file path at which to generate the 3 certificate files of type .pfx, .b64, and .pem. The default is './winrmcert'.",
+        default: "winrmcert"
+
+      option :key_length,
+        short: "-k LENGTH",
+        long: "--key-length LENGTH",
+        description: "Default is 2048",
+        default: "2048"
+
+      option :cert_validity,
+        short: "-cv MONTHS",
+        long: "--cert-validity MONTHS",
+        description: "Default is 24 months",
+        default: "24"
+
+      option :cert_passphrase,
+        short: "-cp PASSWORD",
+        long: "--cert-passphrase PASSWORD",
+        description: "Password for certificate."
```

Comment:
Should this be passphrase to match?

```suggestion
        description: "Passphrase for certificate."
```